includes/AcquiaLiftReportFactory.inc

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * Factory class to create a report object for Acquia Lift.
+ */
+class AcquiaLiftReportFactory {
+
+  /**
+   * Creates a Report object.
+   *
+   * @param $agent_instance
+   *   The Acquia Lift agent that the reporting is for.
+   * @param $api_instance
+   *   The Acquia Lift API classes in use.
+   * @return
+   *   The AcquiaLiftReport class to use.
+   */
+  public static function create(AcquiaLiftLearningAgentInterface $agent_instance, $api_instance, $options) {
+    // Load all reporting-related classes.
+    $report_class_files = array(
+      'AcquiaLiftReportBase',
+      'AcquiaLiftReport',
+      'AcquiaLiftReportCacheInterface',
+      'AcquiaLiftReportCache',
+      'AcquiaLiftReportDataFromFile',
+    );
+    foreach ($report_class_files as $file) {
+      module_load_include('inc', 'acquia_lift', 'includes/' . $file);
+    }
+    $agent_type = $agent_instance->getType();
+    switch ($agent_type) {
+      case ACQUIA_LIFT_TESTING_AGENT_V2:
+        $class = 'AcquiaLiftLearnReport';
```

Comment:
Right, I was going to change this here to not include that stuff but I left it in for the sake of merge conflict avoidance. It won't affect anything seeing as it's not currently possible to create a V2 agent.